pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -259,16 +254,16 @@ func (m *Metadata) EnsureVPCIsPermittedNetwork(ctx context.Context, vpcName stri
 	}
 
 	if !isVPCPermittedNetwork {
-		vpc, err := m.client.GetVPCByName(ctx, vpcName)
+		vpc, err := m.sessionClient.GetVPCByName(ctx, vpcName)
```

Comment:
Can you add a comment in this block that you are relying on the fact that `IsVPCPermittedNetwork` has initialized `sessionClient`? That should make anyone touching this code think twice before removing a function call or changing the order of things.

Reply:
Another option is to do the `client()` call here as well, just in case.

Reply:
Fair point. We could do `client()` but then we'd need to do mutex lock as well, and that could require some code re-ordering/juggling, especially when the calling func is lengthy and complex, and I thought I'd rather leave it as is, directly accessing `.sessionClient`, assuming its (magically) there.

Reply:
I was able to replace `m.sessionClient` direct access with mutex lock and `m.client()`, making the PR a bit bigger, but a private cluster still comes up in PowerVS! PTAL, @r4f4.

Reply:
A bit bigger but even safer so big +1 to that.